### PR TITLE
fix: controlled behavior of navigation in toolbar

### PR DESCRIPTION
**Pull request by fralongo**
### Description

- Focus management when navigation opens
- force closure of the navigation in mobile
- activation of additional mobile tests

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/app-layout/visual-refresh-toolbar/index.tsx [hunk 1]
```diff
@@ -87,6 +83,23 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
...
+    const [navigationOpen = false, setNavigationOpen] = useControllable(
+      controlledNavigationOpen,
+      onNavigationChange,
+      false,
+      {
+        componentName: 'AppLayout',
+        controlledProp: 'navigationOpen',
+        changeHandler: 'onNavigationChange',
+      }
+    );
```
Comment (just-boris) [inline src/app-layout/visual-refresh-toolbar/index.tsx:95]: This is already happening in the shared code for all app layout versions: https://github.com/cloudscape-design/components/blob/54d2d7413c8fcd20d8776e15fedbc0f95ee94ee5/src/app-layout/index.tsx#L94-L99

Not needed here

Comment (fralongo) [reply src/app-layout/visual-refresh-toolbar/index.tsx:95]: Somehow without this the focus management does not work: clicking on the navigation toggle does not focus on the close button of the open navigation, and viceversa. Any other hint?

Comment (just-boris) [reply src/app-layout/visual-refresh-toolbar/index.tsx:95]: No hints, sadly, but double useControllable is definitely not a solution

File: src/app-layout/__tests__/mobile.test.tsx [hunk 2]
```diff
@@ -59,15 +61,24 @@ describeEachAppLayout({ sizes: ['mobile'] }, ({ theme }) => {
...
         : mobileToolbarStyles['mobile-bar'];
-  const drawerBarClassName = theme.startsWith('refresh')
-    ? visualRefreshRefactoredStyles['drawers-mobile-triggers-container']
-    : drawersMobileStyles['drawers-container'];
+  const drawerBarClassName =
+    theme === 'refresh'
+      ? visualRefreshRefactoredStyles['drawers-mobile-triggers-container']
+      : theme === 'refresh-toolbar'
+        ? toolbarStyles['drawers-trigger-content']
+        : drawersMobileStyles['drawers-container'];
```
Comment (just-boris) [inline src/app-layout/__tests__/mobile.test.tsx:69]:
```suggestion
const drawerBarClassName = {
   'classic': drawersMobileStyles['drawers-container'],
   'refresh': visualRefreshRefactoredStyles['drawers-mobile-triggers-container'],
   'refresh-toolbar': toolbarStyles['drawers-trigger-content']
}[theme];
```

we can also make it type-safe to ensure none of the themes are missing

**Review by just-boris — DISMISSED** [review]

File: src/app-layout/visual-refresh-toolbar/index.tsx (hunk 1 above)
Comment (fralongo) [reply src/app-layout/visual-refresh-toolbar/index.tsx]: Solved

**Review by just-boris — APPROVED** [review]

**Pull request merged**